internal/sshtunneler/authentication_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshtunneler
+
+import (
+	"encoding/base64"
+	"time"
+
+	jc "github.com/juju/testing/checkers"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	gomock "go.uber.org/mock/gomock"
+	gc "gopkg.in/check.v1"
+)
+
+type authenticationSuite struct {
+	clock *MockClock
+}
+
+var _ = gc.Suite(&authenticationSuite{})
+
+func (s *authenticationSuite) setupMocks(c *gc.C) *gomock.Controller {
+	ctrl := gomock.NewController(c)
+
+	s.clock = NewMockClock(ctrl)
```

Comment:
why not use github.com/juju/clock/testclock? we'd have less generated code just for tests..

Reply:
Didn't know about that. I've given it a try, and the mock feels better.
1. With the mock I can make assertions that the Now() function is called when I expect it.
2. Setting the time with the mock is just as easy as with the test clock.